### PR TITLE
Blog app: Add Authorization rules

### PR DESCRIPTION
**Pull request by muhammedb8826**
###  In this PR we worked on the followings:

- A user can delete a post if it is their post
- A user can delete a post if a user is an admin
- A user can delete a comment if the user is the author
- A user can delete a comment if the user is an admin
Note: In order to complete the signing-up process, Find the email confirmation link in the terminal(rails server log) and Click on it. After that you will be able to login

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    can :read, :all
+    can :create, [Post, Comment]
+
+    if user.role == 'admin'
+      can :manage, :all
+    else
+      can :destroy, Post, author_id: user.id
+      can :destroy, Comment, author_id: user.id
+    end
```
Comment (abdulhamiid) [inline app/models/ability.rb:17]: After checking your code, everything looks good 👍 However I'm unable to test it out to ensure that the code work as expected with different users

After signing up, I tried to log in but the `message` shown below is blocking access to the app

`
You have to confirm your email address before continuing.
`

However, I did not receive any confirmation instruction 🤷‍♀️

image

Please, let's solve this issue 🛠👍

**Review by abdulhamiid — CHANGES_REQUESTED** [review]
# Status: Required Changes :recycle:
Hi Team 👋, 
Alt Text
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights 🌟

- ✔️ Descriptions are descriptive
- ✔️ Add authorization rules.
- ✔️ Meaningful commit messages

## Required Changes ♻️
_Please kindly check the comments under the review._

## Optional suggestions 🤷‍♂️
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

## Important Notes 📝
- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (getssh) [reply app/models/ability.rb:17]: Hello @abdulhamiid , after signing up you will get a confirmation link in the terminal (rails server log) and you have to click on it in order to confirm your email address. It was one of the requirements to include email confirmation on signing up. Thank you

**Review by Ismailco — APPROVED** [review]
### Hi @muhammedb8826, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

cr-approve


## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**